src/components/Tool/FixPhoneNumbers/FixPhoneNumbers.tsx

```diff
@@ -202,234 +206,173 @@ const FixPhoneNumbers: React.FC<Props> = ({
     });
   };
 
-  const updatePhoneNumber = async (
-    values: FormValues,
-    personId: string,
-    personIndex: number,
-  ): Promise<void> => {
-    const attributes = [
-      {
-        phoneNumbers: values.people[personIndex].phoneNumbers.nodes.map(
-          (phoneNumber: PersonPhoneNumberFragment) => ({
-            id: phoneNumber.id,
-            primary: phoneNumber.primary,
-            number: phoneNumber.number,
-            validValues: true,
-          }),
-        ),
-        id: personId,
-      },
-    ];
+  const handleSingleConfirm = async (
+    person: PersonInvalidNumberFragment,
+    numbers: PhoneNumber[],
+  ) => {
+    const personName = `${person.firstName} ${person.lastName}`;
+    const phoneNumbers = [] as PersonPhoneNumberInput[];
+    numbers.map((phoneNumber) => {
+      phoneNumbers.push({
+        number: phoneNumber.number,
+        id: phoneNumber.id,
+        primary: phoneNumber.primary,
+        validValues: true,
+      });
+    });
 
     await updateInvalidPhoneNumbers({
       variables: {
         input: {
           accountListId,
-          attributes,
+          attributes: [
+            {
+              id: person.id,
+              phoneNumbers,
+            },
+          ],
         },
       },
       update: (cache) => {
-        cache.evict({ id: `Person:${personId}` });
+        cache.evict({ id: `Person:${person.id}` });
+        cache.gc();
+      },
+      onCompleted: () => {
+        enqueueSnackbar(
+          t(`Successfully updated phone numbers for ${personName}`),
+          {
+            variant: 'success',
+          },
+        );
       },
       onError: () => {
-        enqueueSnackbar(t('Error updating phone numbers'), {
+        enqueueSnackbar(t(`Error updating phone numbers for ${personName}`), {
           variant: 'error',
         });
       },
-      onCompleted: () => {
-        enqueueSnackbar(t('Phone numbers updated!'), {
-          variant: 'success',
-        });
-      },
     });
   };
 
-  const handleDelete = (
-    values: FormValues,
-    setValues: (values: FormValues) => void,
-  ): void => {
-    const temp = JSON.parse(JSON.stringify(values));
-
-    const deleting = temp?.people[
-      deleteModalState.personIndex
-    ].phoneNumbers?.nodes.splice(deleteModalState.numberIndex, 1)[0];
-
-    deleting.destroy = true;
-
-    if (
-      deleting.primary &&
-      temp?.people[deleteModalState.personIndex]?.phoneNumbers?.nodes.length
-    ) {
-      temp.people[deleteModalState.personIndex].phoneNumbers.nodes[0].primary =
-        true;
+  const handleChangePrimary = (personId: string, numberIndex: number): void => {
+    const temp = { ...dataState };
```

Comment:
If `!temp[personId]`, there's no need to clone the state and set it without changing it.

You can add `if (!temp[personId]) { return; }` to the top of the function and remove the `if` block on line 258.